camayoc/tests/qpc/ui/utils.py

```diff
@@ -192,15 +193,20 @@ def create_source(view, credential_name, source_type, source_name, addresses):
 
 def delete_source(view, source_name):
     """Delete a source through the UI."""
-    time.sleep(0.2)  # animation timing wait
+    clear_toasts(view=view)
+    dash = DashboardView(view)
+    dash.nav.select('Sources')
+    time.sleep(0.5)
+    view.wait_for_element(locator=Locator(
+        xpath=(f'//div[text()="{source_name}"]/ancestor::node()[7]'
+               '//*[contains(@class,"pficon-delete")]')))
     GenericLocatorWidget(view, locator=Locator(
-        xpath=f"""//div[text()="{source_name}"]/ancestor::node()[7]
-            //*[contains(@class,"pficon-delete")]""")).click()
+        xpath=(f'//div[text()="{source_name}"]/ancestor::node()[7]'
+               '//*[contains(@class,"pficon-delete")]'))).click()
```

Comment:
@ironfroggy do you know of anyway we could work around this hard coding the index in for this? I'm no expert. It just looks scary/fragile to me

Reply:
its hard to say without more understanding of the structure that's being tested against and the semantics there, which I don't have without much exposure to the project.

Plus, since this selector didn't really change from the previous version, I wouldn't be pressed to change it at the moment.